core/platform/src/syscalls_impl.rs

```diff
@@ -1,17 +1,40 @@
 //! Implements `Syscalls` for all types that implement `RawSyscalls`.
 
-use crate::{return_variant, RawSyscalls, Syscalls, YieldType};
+use crate::{RawSyscalls, Syscalls, YieldNoWaitReturn};
+
+mod yield_op {
+    pub const NO_WAIT: u32 = 0;
+    pub const WAIT: u32 = 1;
+}
 
 impl<S: RawSyscalls> Syscalls for S {
     // -------------------------------------------------------------------------
     // Yield
     // -------------------------------------------------------------------------
 
-    fn yield_wait() {
-        Self::raw_yield(YieldType::Wait);
+    fn yield_no_wait() -> YieldNoWaitReturn {
+        unsafe {
+            // flag can be uninitialized because it is not read before the yield
+            // system call, and the kernel promises to only write to it (not
+            // read it).
+            let mut flag = core::mem::MaybeUninit::<YieldNoWaitReturn>::uninit();
```

Comment:
In the kernel we have been trying to minimize the scope of `unsafe` blocks to just the code that requires it. In that vein, I think this could be moved outside the unsafe block (not that doing so is super important).

Reply:
I'm not sure why I did that, I must've thought that `MaybeUninit::uninit()` is `unsafe`. I moved it outside the `unsafe` block, and adjusted the comments accordingly.